multimedia/VLC/Portfile

```diff
@@ -40,434 +30,673 @@ subport             lib${name} {
+    patchfiles-append           patch-for-lua52.diff
 
-    configure.args-delete   --disable-macosx \
-                            --enable-samplerate \
-                            --disable-realrtsp
-    configure.args-append   --enable-macosx 
-    # taken from VLC's own configure.sh script for OS X:
-    configure.args-append   --disable-samplerate \
-                            --enable-merge-ffmpeg \
-                            --enable-realrtsp
-}
+    platform macosx {
+        default_variants-append +quartz +qtkit
 
-# FreeRDP currently doesn't build with CMake >= 3.1 (#47389)
-variant freerdp description {Build the FreeRDP support; currently dysfunctional} {
-    depends_lib-append          port:FreeRDP
-    # access/rdp.c:45:11: fatal error: 'freerdp/version.h' file not found
-    configure.cppflags-append   -DFREERDP_INTERFACE_VERSION \
-                                -DFREERDP_VERSION_MAJOR=1 \
-                                -DFREERDP_VERSION_MINOR=1
-    configure.args-replace      --disable-freerdp --enable-freerdp
-}
+        configure.args-replace  --disable-macosx-eyetv --enable-macosx-eyetv
 
-variant huge \
-    requires jack shout svg \
-    description {Enable all variants except quartz, smb, freerdp and x11} {}
-
-if {${subport} ne "lib${name}"} {
-    variant full \
-        requires huge quartz smb x11 \
-        description {Enable all variants except x11 (and except freerdp, currently)} {}
-} else {
-    variant full \
-        requires huge quartz smb \
-        description {Enable all variants (except freerdp, currently)} {}
+        if {${subport} ne "lib${name}"} {
+            configure.args-replace  --disable-macosx-vlc-app --enable-macosx-vlc-app
+        }
+
+        post-patch {
+            reinplace "s/Appkit/AppKit/" ${worksrcpath}/configure.ac
+            reinplace "/Sparkle.framework/d" \
+                ${worksrcpath}/extras/package/macosx/vlc.xcodeproj/project.pbxproj
+            reinplace "/SDKROOT/d" \
+                ${worksrcpath}/extras/package/macosx/vlc.xcodeproj/project.pbxproj
+            reinplace "/Growl.framework/d" \
+                ${worksrcpath}/extras/package/macosx/package.mak
+
+            reinplace "s:LD_LIBRARY_PATH:DYLD_LIBRARY_PATH:g" \
+                ${worksrcpath}/Makefile.am
+
+            reinplace "/argv/s/environ/*_NSGetEnviron()/" \
+                ${worksrcpath}/modules/misc/inhibit/xdg.c \
+                ${worksrcpath}/modules/stream_filter/decomp.c
+
+            reinplace "s/extern char \\*\\*environ;/#include <crt_externs.h>/" \
+                ${worksrcpath}/modules/misc/inhibit/xdg.c \
+                ${worksrcpath}/modules/stream_filter/decomp.c
+
+            if {![file exists ${worksrcpath}/contrib/BGHUDAppKit.framework]} {
+                ln -s ${frameworks_dir}/BGHUDAppKit.framework ${worksrcpath}/contrib/BGHUDAppKit.framework
+            }
+
+            # To trick configure
+            file mkdir "${worksrcpath}/contrib/lib"
```

Comment:
This is another bit of code that I didn't write, and I'm frankly not very motivated to spend time to convert these into patchfiles - and keep the programmatic version around because a next VLC release is likely to break the patch but not the search/replace action...

As far as I'm concerned VLC has a build system that hangs together with hatpins and zipties, the Portfile has never been much better, and I'm only maintaining it because I need libVLC (and that already costs me more than the minimal effort I'd like to put into it).